convert2rhel/backup.py

```diff
@@ -352,7 +352,16 @@ def backup(self):
                 shutil.copy2(self.filepath, BACKUP_DIR)
             except (OSError, IOError) as err:
                 # IOError for py2 and OSError for py3
-                loggerinst.critical("Error(%s): %s" % (err.errno, err.strerror))
+
+                # Replace this with loggerinst.critical_no_exit() once that is
+                # written.
+                loggerinst.critical_no_exit("Error(%s): %s" % (err.errno, err.strerror))
+                raise exceptions.CriticalError(
+                    id_="FAILED_TO_SAVE_FILE_TO_BACKUP_DIR",
+                    title="Failed to copy file to the backup directory.",
+                    description="While convert2rhel tests whether it is likely to succeed, spme files would be changed. To enable rollback in case we detect something that we would fail to do, we backup those files prior to changing them. In this case, we encountered a failure while performing that backup so it is unsafe to continue.",
+                    diagnosis="Failed to backup %s. Errno: %s, Error: %s" % (self.filepath, err.errno, err.strerror),
```

Comment:
@mportman12 Hey Miriam! This is the first of several PRs that we will be opening to add messages that will show up in insights.  If you'd care to review it to make sure we're being clear, please do.  I've also added a note to the jira Epic tracking this work to CC you on the PRs which update the messages (If you have a different procedure you'd like to follow when you review messages destined for insights, we can remove that and teach people about that).

Reply:
@abadger, the rest of the messages will be added in this PR or another one?

Reply:
The rest of hte messages should be added in other PRs.  The idea is that this one gives us the Exception and logger methods that we'll need and a single logger.critical ported over so that we have an example.  Then other messages can be split up over multiple people and multiple PRs (using Aleksandra's proposal for large changes.)

Reply:
Gotcha. And they are expected to land in 1.5 or 1.6?